### ISSUE TITLE
Deprecation causes breaking changes in azuread.ApplicationPassword

### ISSUE THREAD

**Issue by Sodki** [pulumi/pulumi-azuread#531]
### What happened?

I've been using `azuread.Application` in conjunction with `azuread.ApplicationPassword` to manage an application. Recently (with pulumi-azuread v5.44.0) the property `application_object_id` of `azuread.ApplicationPassword` has been deprecated in favour of `application_id`. Unfortunately this breaks the workflow, since it now expects the value to be in the format `/applications/<application_id>`, but if we make this change then the resource will try to recreate itself. 

### Example

First start with this code. Notice the deprecation error.

```
app = azuread.Application(
        "example",
        display_name="example",
)

self._app_password = azuread.ApplicationPassword(
        "example",
        application_object_id=app.id,
)
````

Then change it to:

```
app = azuread.Application(
        "example",
        display_name="example",
)

self._app_password = azuread.ApplicationPassword(
        "example",
        application_id=app.id,
)
````

This breaks the code with:

```
parsing the Application ID: the number of segments didn't match
    
    Expected a Application ID that matched (containing 2 segments):
    
    > /applications/00000000-0000-0000-0000-000000000000
    
    However this value was provided (which was parsed into 0 segments):
```

If we add the `/applications/` suffix, then it will try to replace the resource.

### Output of `pulumi about`

```
CLI          
Version      3.91.1
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME          VERSION
azuread       5.44.0
python        unknown

Host     
OS       fedora
Version  38
Arch     x86_64

This project is written in python: executable='/usr/bin/python3' version='3.11.6'

Dependencies:
NAME                 VERSION
pip                  23.3.1
pulumi-azuread       5.44.0
setuptools           68.2.2
wheel                0.41.3
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by tom-fletcher** [comment]
Also experienced this with `azuread.ApplicationFederatedIdentityCredential`. For anyone who wants to get rid of the deprecation warnings and requires an immediate workaround that doesn't replace the resource, you can achieve this with a manual edit of your stack.

**⚠️ Manually editing your stack means you need to know what you are doing, take backups and all due care!**

In your pulumi program, update the resource as follows:

```diff
new azuread.ApplicationFederatedIdentityCredential(
  'resourceName',
    {
-     applicationObjectId: application.objectId,
+     applicationId: application.id,
      ...
  }
);
```

Export the stack using `pulumi stack export --file stack.json`, and find the relevant resource. Update the `inputs` to reflect the requirement for the `/applications` prefux

```diff
"inputs": {
-   "applicationObjectId": "00000000-0000-0000-0000-000000000000",
+   "applicationId": "/applications/00000000-0000-0000-0000-000000000000",
    ...
},
```

Export your stack using `pulumi stack export --file stack.json`.

This will allow you to do `pulumi up` again without any issues.

**Comment by mikhailshilkov** [comment]
@Sodki What is the format of `app.id` in your stack?

I'm asking because I ran your repro from scratch and had no issue updating from `applicationObjectId` to `applicationId`. I observed that `app.id` in my stack is `/applications/d6ae21f8-67fd-4eb0-94b8-3f953a32f031` so it matches the required format and update is a no-op. Is it different for you them? What if you run your program from scratch in a new stack (as an experiment)?
